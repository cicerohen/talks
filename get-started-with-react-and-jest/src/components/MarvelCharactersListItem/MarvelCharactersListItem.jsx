import React from 'react';
import PropTypes from 'prop-types';
import ListItemWithAvatar from '../ListItemWithAvatar';
import listItemWithAvatarStyle from '../ListItemWithAvatar/style/ListItemWithAvatar.scss';
import Style from './style/MarvelCharactersListItem.scss';

const MarvelCharactersListItem = ({ name, description, thumbnail }) => (
  <ListItemWithAvatar
    avatarUrl={thumbnail}
    style={{
      ...listItemWithAvatarStyle,
      avatar__root: Style.avatar__root,
    }}
  >
    <p>{name}</p>
    <p>{description}</p>
  </ListItemWithAvatar>
);

MarvelCharactersListItem.propTypes = {
  name: PropTypes.string.isRequired,
  thumbnail: PropTypes.string.isRequired,
  description: PropTypes.string,
};

MarvelCharactersListItem.defaultProps = {
  description: '',
};

export default MarvelCharactersListItem;

