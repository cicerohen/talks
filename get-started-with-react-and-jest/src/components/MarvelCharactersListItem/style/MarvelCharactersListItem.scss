.avatar__root {
  composes: avatar__root from 'components/ListItemWithAvatar/style/ListItemWithAvatar.scss';
  width: 80px;
  height: 80px;
}