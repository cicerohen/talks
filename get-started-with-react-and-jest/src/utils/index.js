import { Iterable, fromJS } from 'immutable';
import { createBrowserHistory } from 'history';

export const asImmutable = obj => (Iterable.isIterable(obj) ? obj : fromJS(obj));
export const emptyMap = asImmutable({});

export const getHistory = () => {
    window.browserHistory = window.browserHistory || createBrowserHistory();
    return window.browserHistory;
};

export const isDevEnv = () => {
  const nodeEnv = process.env.NODE_ENV;
  return nodeEnv !== 'production' && nodeEnv !== 'test';
};
