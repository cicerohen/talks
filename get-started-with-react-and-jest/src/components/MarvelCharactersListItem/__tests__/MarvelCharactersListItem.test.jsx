import React from 'react';
import { shallow } from 'enzyme';
import { asImmutable } from '../../../utils';
import MarvelCharactersListItem from '../MarvelCharactersListItem';

const minimalProps = {
  name: 'character name',
  thumbnail: 'character thumb',
}

const commonProps = {
  ...minimalProps,
  description: 'character desc',
}

describe('<MarvelCharactersListItem /> Tests', () => {
  it('should render without exploding', (done) => {
    const wrapper = shallow(
      <MarvelCharactersListItem {...minimalProps}/>
    );
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly', (done) => {
    const wrapper = shallow(
      <MarvelCharactersListItem {...minimalProps}/>
    );
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should render correctly with all props', (done) => {
    const wrapper = shallow(
      <MarvelCharactersListItem {...commonProps}/>
    );
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
