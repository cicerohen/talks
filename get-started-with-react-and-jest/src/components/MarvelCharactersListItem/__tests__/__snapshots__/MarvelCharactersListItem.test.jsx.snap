// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarvelCharactersListItem /> Tests should render correctly 1`] = `
<ListItemWithAvatar
  avatarUrl="character thumb"
  style={
    Object {
      "avatar__root": "avatar__root",
    }
  }
>
  <p>
    character name
  </p>
  <p />
</ListItemWithAvatar>
`;

exports[`<MarvelCharactersListItem /> Tests should render correctly with all props 1`] = `
<ListItemWithAvatar
  avatarUrl="character thumb"
  style={
    Object {
      "avatar__root": "avatar__root",
    }
  }
>
  <p>
    character name
  </p>
  <p>
    character desc
  </p>
</ListItemWithAvatar>
`;
